src/event_listener/shared.rs

```diff
@@ -666,109 +669,307 @@ enum ParsedEventType {
-            ),
-            (
-                ParsedEventType::WorkspaceMoved,
-                r"moveworkspace>>(?P<workspace>.*),(?P<monitor>.*)"
-            ),
-            (
-                ParsedEventType::WorkspaceRename,
-                r"renameworkspace>>(?P<id>.*),(?P<name>.*)"
-            ),
-            (
-                ParsedEventType::ActiveMonitorChanged,
-                r"focusedmon>>(?P<monitor>.*),(?P<workspace>.*)"
-            ),
-            (
-                ParsedEventType::ActiveWindowChangedV1,
-                r"activewindow>>(?P<class>.*?),(?P<title>.*)"
-            ),
-            (
-                ParsedEventType::ActiveWindowChangedV2,
-                r"activewindowv2>>(?P<address>.*)"
-            ),
-            (
-                ParsedEventType::FullscreenStateChanged,
-                r"fullscreen>>(?P<state>0|1)"
-            ),
-            (
-                ParsedEventType::MonitorRemoved,
-                r"monitorremoved>>(?P<monitor>.*)"
-            ),
-            (
-                ParsedEventType::MonitorAdded,
-                r"monitoradded>>(?P<monitor>.*)"
-            ),
-            (
-                ParsedEventType::WindowOpened,
-                r"openwindow>>(?P<address>.*),(?P<workspace>.*),(?P<class>.*),(?P<title>.*)"
-            ),
-            (
-                ParsedEventType::WindowClosed,
-                r"closewindow>>(?P<address>.*)"
-            ),
-            (
-                ParsedEventType::WindowMoved,
-                r"movewindow>>(?P<address>.*),(?P<workspace>.*)"
-            ),
-            (
-                ParsedEventType::LayoutChanged,
-                r"activelayout>>(?P<keyboard>.*)(?P<layout>.*)"
-            ),
-            (ParsedEventType::SubMapChanged, r"submap>>(?P<submap>.*)"),
-            (
-                ParsedEventType::LayerOpened,
-                r"openlayer>>(?P<namespace>.*)"
-            ),
-            (
-                ParsedEventType::LayerClosed,
-                r"closelayer>>(?P<namespace>.*)"
-            ),
-            (
-                ParsedEventType::FloatStateChanged,
-                r"changefloatingmode>>(?P<address>.*),(?P<floatstate>[0-1])"
-            ),
-            (
-                ParsedEventType::Minimize,
-                r"minimize>>(?P<address>.*),(?P<state>[0-1])"
-            ),
-            (
-                ParsedEventType::Screencast,
-                r"screencast>>(?P<state>[0-1]),(?P<owner>[0-1])"
-            ),
-            (
-                ParsedEventType::UrgentStateChanged,
-                r"urgent>>(?P<address>.*)"
-            ),
-            (
-                ParsedEventType::WindowTitleChanged,
-                r"windowtitle>>(?P<address>.*)"
-            ),
-            (ParsedEventType::Unknown, r"(?P<Event>^[^>]*)"),
-        ]
+/// All the recognized events
+static EVENT_SET: Lazy<[(ParsedEventType, Regex); 24]> = Lazy::new(|| {
```

Comment:
I made it an array because slice size is computed at runtime every time. I will probably change it to a slice though because if you have a problem with changing the hardcoded length every time, then a slice is probably the best option, since the perf difference is negligible.

Reply:
Upon actually trying it, I found I can't just have a slice. I have to either have a Boxed slice, or an array.